src/EA.Weee.Database/scripts/Update/Release 7/Sprint7/20241105-1141-producer-register-report-returned.sql

```diff
@@ -0,0 +1,231 @@
+    INNER JOIN Producer.RegisteredProducer RP ON PS.RegisteredProducerId = RP.Id
+    WHERE MU.IsSubmitted = 1 AND RP.Removed = 0
+) P_First ON PS.RegisteredProducerId = P_First.RegisteredProducerId AND P_First.RowNumber = 1
+
+-- Partners subquery
+LEFT JOIN (
+    SELECT DISTINCT 
+        P.Id, 
+        STUFF((
+            SELECT DISTINCT '; ' + PP.Name
+            FROM Producer.Partner PP 
+            WHERE P.Id = PP.PartnershipId
+            FOR XML PATH(''), TYPE
+        ).value('.', 'NVARCHAR(MAX)'), 1, 2, '') AS Partners
+    FROM Producer.Partnership P
+) Partners ON PBP.Id = Partners.Id
+
+WHERE
+    MU.ComplianceYear = @ComplianceYear
+    AND RP.Removed = 0
+
+UNION ALL
+
+	-- Query for the direct registrant dataset
+    SELECT
+		o.[Name] AS 'CompanyName',
+		COALESCE(ap.OverseasProducerName, o.[Name], '') AS 'ProducerName',
+		COALESCE(ap.OverseasProducerTradingName, o.[TradingName], '') AS 'TradingName',
+		dbo.GetObligationType(eorv.Id) AS 'ObligationType',
+		-- Registered Office Details
+        oa.Address1 as 'ROAPrimaryName',
+        NULL as 'ROASecondaryName',
+        oa.Address2 as 'ROAStreet',
+        oa.TownOrCity as 'ROATown',
+        oa.CountyOrRegion as 'ROALocality',
+        NULL as 'ROAAdministrativeArea',
+        oa.PostCode as 'ROAPostCode',
+        loc.Name as 'ROACountry',
+		oa.Telephone as 'ROATelephone',
+        oa.Email as 'ROAEmail',
+		rp.ProducerRegistrationNumber AS 'PRN',
+		'Direct registrant' AS SchemeName,
+		NULL AS 'SchemeOperator',
+		NULL AS 'CSROAAddress1',
+		NULL AS 'CSROAAddress2',
+		NULL AS 'CSROATownOrCity',
+		NULL AS 'CSROACountyOrRegion',
+		NULL AS 'CSROAPostcode',
+		NULL AS 'CSROACountry',
+		ap.OverseasProducerName as 'OPNAName',
+		pa.PrimaryName as 'OPNAPrimaryName',
+        pa.SecondaryName as 'OPNASecondaryName',
+		pa.Street as 'OPNAStreet',
+        pa.Town as 'OPNATown',
+        pa.Locality as 'OPNALocality',
+        pa.AdministrativeArea as 'OPNAAdministrativeArea',
+		ac.Name as 'OPNACountry',
+        pa.PostCode as 'OPNAPostCode',
+		dps.ComplianceYear,
+		NULL AS 'PPOBPrimaryName',
+		NULL AS 'PPOBSecondaryName',
+		NULL AS 'PPOBStreet',
+		NULL AS 'PPOBTown',
+		NULL AS 'PPOBLocality',
+		NULL AS 'PPOBAdministrativeArea',
+		NULL AS 'PPOBCountry',
+		NULL AS 'PPOBPostcode',
+		1 AS IsDirectProducer
+	FROM
+        [Producer].[DirectProducerSubmission] dps
+        INNER JOIN [Producer].[DirectRegistrant] dr ON dr.Id = dps.DirectRegistrantId
+        INNER JOIN [Organisation].[Organisation] o ON o.Id = dr.OrganisationId
+        INNER JOIN [Organisation].[Address] oa ON oa.Id = o.BusinessAddressId
+		INNER JOIN [Organisation].[Contact] oc ON oc.Id = dr.ContactId
+        INNER JOIN [Lookup].[Country] loc ON loc.Id = oa.CountryId
+        INNER JOIN [Producer].[RegisteredProducer] rp ON dps.RegisteredProducerId = rp.Id AND dps.ComplianceYear = @ComplianceYear
+        INNER JOIN (
+			SELECT 
+				DirectProducerSubmissionId,
```

Comment:
Change this to join on most recently submitted rather than the current submission.